tests/test_utils.py

```diff
@@ -812,4 +812,55 @@ def test_validation_coordinates(self):
 
         # test for strings
         with pytest.raises(ValueError, match=f"could not convert string to float: 'test'"):
-            Utils.validate_coordinates([(0.0, 0.0, 0.0), (0.0, "test", 0.0)])
\ No newline at end of file
+            Utils.validate_coordinates([(0.0, 0.0, 0.0), (0.0, "test", 0.0)])
+
+    def test_create_sleeper_volume_at_origin(self):
+        """
+        Test the creation of a sleeper volume at the origin.
+
+        This test ensures that the function correctly calculates the sleeper volume
+        when the origin is at (0, 0, 0).
+
+        Asserts that the result matches the expected output.
+        """
+        sleeper_rail_pad_offset = 0.5
+        local_coord = [0.0, 0.0, 0.0]
+        sleeper_dimensions = [2.0, 4.0, 1.0]  # length, width, height
+        expected = np.array([[1.5, 0.0, 2.0], [1.5, 0.0, -2.0], [-0.5, 0.0, -2.0], [-0.5, 0.0, 2.0]])
+        result = Utils.create_sleeper_volume(local_coord, sleeper_dimensions, sleeper_rail_pad_offset)
+        np.testing.assert_array_almost_equal(result, expected)
+
+    def test_create_sleeper_volume_nonzero_origin(self):
+        """
+        Test the creation of a sleeper volume with a non-zero origin.
+
+        This test ensures that the function correctly calculates the sleeper volume
+        when given a non-zero origin.
+
+        Asserts that the result matches the expected output.
+        """
+        local_coord = [1.0, -1.0, 0.5]
+        sleeper_dimensions = [3.0, 2.0, 0.5]
+        sleeper_rail_pad_offset = 0.5
+        expected = np.array([[3.5, -1.0, 1.5], [3.5, -1.0, -0.5], [0.5, -1.0, -0.5], [0.5, -1.0, 1.5]])
```

Comment:
also here, all the vertical coordinates are equal, therefore this is not a volume

Reply:
Changed name of function as it does not retrun the volume but the base coordinates